tests/TestCase.php

```diff
@@ -250,7 +243,8 @@ protected function getInvalidBooleanValues(bool $includeNull = false): array
      */
     protected function getInvalidDocumentValues(bool $includeNull = false): array
     {
-        return array_merge([123, 3.14, 'foo', true, PackedArray::fromPHP([])], $includeNull ? [null] : []);
+        return array_merge([123, 3.14, 'foo', true], $includeNull ? [null] : []);
+//        return array_merge([123, 3.14, 'foo', true, PackedArray::fromPHP([])], $includeNull ? [null] : []);
```

Comment:
You left this commented.

Reply:
Whoops, removed.

For context, with the `object|array` type now enforced for documents, `PackedArray` instances no longer cause an error. I could bring back the `PackedArray` check if we want to be more specific about what objects are allowed, but I defer to @jmikola on this.

Reply:
Isn't the PackedArray check still in place? I saw you removed various conditionals when adding `array|object` type hints, but `is_document()` still rejects PackedArray instances. I would expect that that `InvalidArgumentException::expectedDocumentType()` still gets thrown for PackedArray instances.

I assume changing `InvalidArgumentException` expectations to `TypeError` is what conflicts here. I don't feel strongly about this, but I'd argue that parameter and option type checks shouldn't necessarily be expected to raise `TypeError`. There are still various other contexts where we raise `InvalidArgumentException` for validation errors (e.g. option value is out of range, or a string value doesn't satisfy our homemade enum).

Reply:
It is indeed. I've updated the type assertions based on the value given to expect a `TypeError` or `InvalidArgumentException` and restored the `PackedArray` instance as an invalid document, as it adds confidence that we have the right checks in place.

With `TypeError` being a thing, I wonder if we should consider throwing `TypeError` throwables instead of `InvalidArgumentException` on type mismatches.

Reply:
> With TypeError being a thing, I wonder if we should consider throwing TypeError throwables instead of InvalidArgumentException on type mismatches.

That sounds like a great idea, now that we can reliably expect TypeErrors from PHP itself. Might be something to consider for both PHPC and PHPLIB, and then we can also remove the vapid documentation for throwing InvalidArgumentException from most things.